src/book.js

```diff
@@ -553,201 +553,49 @@ export class Book {
     const papersize = config.papersize;
     const outPDF = config.outPDF;
     const positions = config.positions;
-    const cropmarks = config.cropmarks;
+    const foldmarks = config.cropmarks;
     const pdfEdgeMarks = config.pdfEdgeMarks;
     const cutmarks = config.cutmarks;
     const alt = config.alt;
     let side2flag = config.side2flag;
 
     const block = config.embeddedPages.slice(block_start, block_end);
-    const currPage = outPDF.addPage([papersize[0], papersize[1]]);
+    const currPage = outPDF.addPage(papersize);
+    const cropLines = cutmarks ? drawCropmarks(papersize, this.per_sheet) : [];
+    const foldLines = foldmarks
+      ? drawFoldlines(side2flag, this.duplexrotate, papersize, this.per_sheet)
+      : [];
+    const drawLines = [...cropLines, ...foldLines];
 
     block.forEach((page, i) => {
       if (page == 'b' || page === undefined) {
         // blank page, move on.
       } else {
-        const pos = positions[i];
-        const rot = pos.rotation;
+        const { y, x, sx, sy, rotation } = positions[i];
         currPage.drawPage(page, {
-          y: pos.y,
-          x: pos.x,
-          xScale: pos.sx,
-          yScale: pos.sy,
-          rotate: degrees(rot),
+          y,
+          x,
+          xScale: sx,
+          yScale: sy,
+          rotate: degrees(rotation),
         });
       }
-    });
-    block.forEach((page, i) => {
```

Comment:
looks like you removed the second block, which I would caution against. If we keep them separate, then pages that may overflow onto each other will be allowed/safe and the markings will be drawn atop them in a second pass. If they're drawn in the sequence of the pages, then the chances of them being over-written is determined by page order.

Reply:
So with the way the code is refactored, it generates the positions for the marks in the loop, but they aren't actually drawn until farther down, where we draw all the lines. Because that's after position block, there shouldn't be any issues with overlapping.

Reply:
!!! Ack! Thank you for pointing that out! Clearly I failed to read closely enough-- this is a better approach 🙏